src/noisepy/seis/S2_stacking.py

```diff
@@ -49,8 +50,49 @@ def stack(
     fft_params: ConfigParameters,
     scheduler: Scheduler = SingleNodeScheduler(),
 ):
+    executor = ProcessPoolExecutor()
     tlog = TimeLogger(logger=logger, level=logging.INFO)
     t_tot = tlog.reset()
+
+    def initializer():
+        timespans = cc_store.get_timespans()
+        pairs_all = cc_store.get_station_pairs()
+        logger.info(f"Station pairs: {pairs_all}")
+
+        if len(timespans) == 0 or len(pairs_all) == 0:
+            raise IOError("Abort! no available CCF data for stacking")
+        return timespans, pairs_all
+
+    timespans, pairs_all = scheduler.initialize(initializer, 2)
+
+    # Get the pairs that need to be processed by this node
+    pairs_node = [pairs_all[i] for i in scheduler.get_indices(pairs_all)]
+
```

Comment:
The main change is here, using the executor to run all the jobs in parallel.

The bulk of the changes below are: a) a change in indentation since most lines went from being inside a loop to being just part of a function, and b) `continue` -> `return`